gatsby-config.js

```diff
@@ -132,15 +134,73 @@ module.exports = {
     {
       resolve: `gatsby-plugin-sitemap`,
       options: {
+        // to read more: https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap/#options
         excludes: SEO_EXCLUDED_URLS,
-        resolvePagePath: ({ path }) => {
-          if (!path.endsWith('/')) {
+        query: `
+        {
+          site {
+             siteMetadata {
+               siteUrl
+             }
+           }
+          allSitePage {
+            nodes {
+              path
+            }
+          }
+        }`,
+        // returns path und translated path of page
+        resolvePages: ({ allSitePage: { nodes: allPages } }) => {
+          return allPages.map((page) => {
+            const translation = allPages.find((translatedPage) => {
+              const path = page.path;
+              const translatedPath = translatedPage.path;
+              if (translatedPath === path) {
+                return false;
+              }
+              if (path.includes('/de/')) {
+                return path.replace('/de', '') === translatedPath;
+              } else {
+                return translatedPath.replace('/de/', '/') === path;
+              }
+            });
+            return {
+              path: page.path,
+              translation: translation ? translation.path : undefined,
+            };
+          });
+        },
+        filterPages: (page, excludes) => {
+          if (excludes === page.path) {
+            return true; // excludes page
+          }
+          if (!page.path.endsWith('/')) {
             console.warn(
               'Path of the page does not end with a slash! For SEO reasons all paths should end with a slash:',
-              path,
+              page.path,
             );
           }
-          return path;
+          return false;
+        },
+        serialize: ({ path, translation }) => {
+          const isGermanPath = path.includes('/de/');
+          if (!translation) {
+            return {
+              url: path,
+              changefreq: 'daily',
+              priority: 0.7,
+              links: [{ lang: isGermanPath ? 'de' : 'en', url: path }],
+            };
+          }
+          return {
+            url: path,
+            changefreq: 'daily',
+            priority: 0.7,
+            links: [
+              { lang: 'de', url: isGermanPath ? path : translation },
+              { lang: 'en', url: isGermanPath ? translation : path },
+            ],
+          };
```

Comment:
This is easily understood. Create a sitemap entry for each language and list all possible languages available. Looks as promised in the output.
![Screen Shot 2021-08-31 at 15 00 06](https://user-images.githubusercontent.com/1701755/131506873-abdd726f-7074-4765-84e2-54d8af305b76.png)

I also double checked the output according to the requirements listed here (https://developers.google.com/search/docs/advanced/crawling/localized-versions#sitemap) and it looks good. Pretty high amount of duplication required. List each page translated or not and for each page translate the available translations but it's simple and easy to check 👍